### PR TITLE
feat: create azure issue

### PR DESCRIPTION
**Pull request by nickoferrall**
Fix https://github.com/ParabolInc/parabol/issues/7026

https://www.loom.com/share/f4a4a7a97e3c4397a5373782cfa230f0

### To test

- Integrate with Azure DevOps. You may need to run `yarn postdeploy` or add the feature flag if it's not an option in your integration table.
- Go to the Scope phase of a Sprint Poker meeting
- Create an issue using the fab
- Can successfully change the project that the issue is sent to 
- The issue shows up at the top of the results
- Can click on the URL in the newly issue and get taken to that issue

Note: 

- The link in the Azure item says "Basic:Issue" rather than the project name. I suggest we change this and have created an issue: https://github.com/ParabolInc/parabol/issues/7048
- If you create many issues and refresh the page a few times, the integration can timeout. I created an issue for this: https://github.com/ParabolInc/parabol/issues/7047
- After ~24 hours, I need to manually refresh the Azure token to see any results: https://github.com/ParabolInc/parabol/issues/7006
- After selecting an issue in the Scope phase or creating a new issue, the subtitle in the newly created item is an id instead of the issue number. Issue here: https://github.com/ParabolInc/parabol/issues/7051

### CODE REVIEW

File: packages/client/components/NewAzureIssueInput.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+interface Props {
+  isEditing: boolean
+  meetingId: string
+  setIsEditing: (isEditing: boolean) => void
+  viewerRef: NewAzureIssueInput_viewer$key
+}
+
+const validateIssue = (issue: string) => {
+  return new Legitity(issue).trim().min(2, `C’mon, you call that an issue?`)
```
Comment (igorlesnenko) [inline packages/client/components/NewAzureIssueInput.tsx:106]: +1 The error message feels not obvious :)

File: packages/client/components/NewAzureIssueInput.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            }
+          }
+        }
+      }
+    `,
+    viewerRef
+  )
+  const {id: userId, team, teamMember} = viewer
+  const {id: teamId} = team!
+  const projects = teamMember?.integrations?.azureDevOps.projects || []
```
Comment (igorlesnenko) [inline packages/client/components/NewAzureIssueInput.tsx:136]: +1 Using `??` is preferable

File: packages/client/components/NewAzureIssueInput.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    const selectedProject = projects.find((project) => project.name === selectedProjectName)!
+    const serviceProjectHash = AzureDevOpsProjectId.join(
+      selectedProject.instanceId,
+      selectedProject.id
+    )
+    const newTask = {
+      teamId,
+      userId,
+      meetingId,
+      content: convertToTaskContent(`${newIssueTitle} #archived`),
```
Comment (igorlesnenko) [inline packages/client/components/NewAzureIssueInput.tsx:182]: Is it expected behaviour? We always creating archived tasks in our board? Do you know the reason behind this?

Comment (nickoferrall) [reply packages/client/components/NewAzureIssueInput.tsx:182]: By creating an archived task, the `integration` is stored in the `task` and can be used to insert it into the Relay connection:
https://github.com/ParabolInc/parabol/blob/8e96045b57741440e39cd19cd8d38883fcbc629b/packages/client/mutations/handlers/handleGitHubCreateIssue.ts#L11

Also, as we already had this logic built for the task integration, I'm guessing it was easier to use the existing logic in `CreateTaskMutation` than create a new mutation (or update `UpdatePokerScopeMutation`) that adds the integration into the connection.

Does that sound right, @mattkrick?

File: packages/client/components/NewAzureIssueInput.tsx (hunk 1 above)
Comment (nickoferrall) [reply packages/client/components/NewAzureIssueInput.tsx:106]: That's a fair point. For me, I prefer using apps that feel playful, and I think informal language helps with that. It's also consistent with other parts of the app - we have 9 uses of the word "C'mon" 😄 

We could allow any length, but it's unlikely that the user meant to create an issue if they haven't hit the two-character requirement, so I think it's a better UX to give them that feedback

File: packages/client/components/NewAzureIssueInput.tsx (hunk 3 above)
Comment (mattkrick) [reply packages/client/components/NewAzureIssueInput.tsx:182]: working as intended? yes
correct? no :disappointed: 
the problem was folks would run a poker meeting, import a bunch of tasks & then when they finish the meeting, those tasks would be on their kanban board. they didn't like this, and I can't blame them, they pointed "stories" but in our system, everything is a "Task". Then again, some users like having stories show up on their task board! How do we make both sides happy?
If someone would like to tackle this problem, it'd give me great joy

**Review by mattkrick — APPROVED** [review]

File: packages/client/components/NewAzureIssueInput.tsx (hunk 3 above)
Comment (nickoferrall) [reply packages/client/components/NewAzureIssueInput.tsx:182]: I see - so the original reason to create a task when creating an integration issue was to point to the story? Why do we need to point to a story when we're querying directly from the API? If we create a GitHub issue from a Poker meeting, we can query that issue directly from GitHub's API without creating a Parabol task?

**Pull request merged**